commons/src/main/java/fr/free/nrw/commons/upload/ImageProcessing.java

```diff
@@ -0,0 +1,125 @@
+package fr.free.nrw.commons.upload;
+
+import android.content.Context;
+import android.database.Cursor;
+import android.media.ExifInterface;
+import android.net.Uri;
+import android.provider.DocumentsContract;
+import android.provider.MediaStore;
+import android.util.Log;
+
+import java.io.IOException;
+
+
+public class ImageProcessing {
+
+    private Uri uri;
+    private Context context;
+
+    public ImageProcessing(Context context, Uri uri){
+        this.context = context;
+        this.uri = uri;
+    }
+    /**
+    * Gets file path of image from its Uri
+    * May return null
+    */
+    public String getFilePath(){
```

Comment:
It feels like this responsibility is something a little different. The class is actually making calls to a ContentProvider. Consider a second class for doing URI to filename conversions.